gef.py

```diff
@@ -5179,12 +5180,36 @@ def do_invoke(self, argv):
 
         pattern = argv[0]
         endian = get_endian()
-        if argc==2:
-            if argv[1]=="big": endian = Elf.BIG_ENDIAN
-            elif argv[1]=="small": endian = Elf.LITTLE_ENDIAN
 
-        info("Searching '{:s}' in memory".format(Color.yellowify(pattern)))
-        self.search_pattern(pattern, endian)
+        if argc >= 2:
+            if argv[1] == "big": endian = Elf.BIG_ENDIAN
+            elif argv[1] == "small": endian = Elf.LITTLE_ENDIAN
+
+        if is_hex(pattern):
+            if endian == Elf.BIG_ENDIAN:
+                pattern = "".join(["\\x"+pattern[i:i+2] for i in range(2, len(pattern), 2)])
+            else:
+                pattern = "".join(["\\x"+pattern[i:i+2] for i in range(len(pattern) - 2, 0, -2)])
+
+        if argc == 3:
+            info("Searching '{:s}' in {:s}".format(Color.yellowify(pattern), argv[2]))
+
+            if "0x" in argv[2]:
+                start, end = parse_string_range(argv[2])
+
+                self.print_section(lookup_address(start).section)
```

Comment:
`lookup_address()` can return an `Address` marked as invalid (`addr.valid = False`). We should test that before using it in the `print_section()` call.